### DISCUSSION TITLE
Enable grouping at the datasource level

### DISCUSSION THREAD

**Discussion by ikesyo** [renovatebot/renovate#22350] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

IIUC, currently, specifying a `groupName` is only possible at the manager level by [`PackageDependency.groupName`](https://github.com/renovatebot/renovate/blob/d9f5d04e210b69c65a686503265ea2b707f31cd6/lib/modules/manager/types.ts#L114) (except manual configurations in `renovate.json`). However, this is not desirable in the case of cpanfile.

In CPAN, packages are uploaded and distributed at the [Distribution](https://perldoc.perl.org/CPAN::Meta::Spec#distribution) level. On the other hand, dependencies specified in the cpanfile are at the [Module](https://perldoc.perl.org/CPAN::Meta::Spec#module) level. A distribution can include multiple modules. When multiple modules from a same distribution are listed in a cpanfile, they should be grouped and updated together. However, since each line in a cpanfile only contains the module name without the distribution name, parsing the cpanfile by the cpanfile manager does not provide enough information for grouping.

In the implementation of the cpan datasource, it is possible to retrieve distribution information along with each module's releases (actually its distributions's releases). I think this problem can be solved by allowing the specification of `groupName` at the datasource level (through `ReleaseResult` and `UpdateResult`). Although I haven't checked the details, it may also be useful for other managers/datasources besides the cpanfile manager/cpan datasource.

- Implementation: #22332

**Comment by rarkins** [comment]
I think it would be better to introduce a generic concept like "scope" (e.g. like `@angular/` in npm, or the "group ID" in maven, and doing grouping by scope - instead of overriding the use of `groupName`

**Comment by ikesyo** [comment]
#11092 ? 

Implementation path for `scope` (is this for manager or/and datasource?) and `groupPreference` seems not obvious yet.

For our usecase, `scope` needs to be set at datasource level, not only at manager level (as described above).
